stores/s3store/store.go

```diff
@@ -0,0 +1,72 @@
+package s3store
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/mailchain/mailchain/encoding"
+	"github.com/pkg/errors"
+)
+
+type S3Store struct {
```

Comment:
This name is not ideal, it creates stutter. `s3store.S3Store` not 100% sure on the best name but `s3store.Uploader` seems ok.